sunscreen_zkp_backend/src/bulletproofs.rs

```diff
@@ -335,13 +335,21 @@ impl Default for BulletproofsBackend {
 
 fn constraint_count(graph: &ExecutableZkpProgram) -> Result<usize> {
     let mut count = 0;
+    let mut input_count = 0usize;
 
     let query = GraphQuery::new(graph);
 
     for i in graph.node_indices() {
         let node = &graph[i];
 
         match node.operation {
+            Operation::Input(_) => {
```

Comment:
I *still* don't quite understand what exactly this algorithm should be. I think this might be overestimating.